resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -1137,6 +1138,26 @@ protected void tearDown() throws Exception {
 		FreezeMonitor.done();
 	}
 
+	private void checkStateOfJobManager() {
+		boolean suspended = Job.getJobManager().isSuspended();
+
+		resumeJobManagerIfNecessary();
+
+		assertFalse("This test stopped the JobManager, which could have affected other tests.", //
+				suspended);
+	}
+
+	private void resumeJobManagerIfNecessary() {
+		if (!Job.getJobManager().isSuspended()) {
+			return;
+		}
+
+		String warningMessage = "The JobManager has been resumed. Please remember to always clean-up yourself after every test "
+				+ "in order to avoid random errors in subsequent tests (i.e. isolate your tests)";
+		TestUtil.log(IStatus.WARNING, getName(), warningMessage);
```

Comment:
In case the `JobManager` was suspended and not resumed, a warning will be logged and the test will fail with an assertion error. Is this necessary or wouldn't it be sufficient to throw an assertion error with this message. Then it would be sufficient to call `jobManager.resume()` in the `checkStateOfJobManager()` method.